jax/numpy/lax_numpy.py

```diff
@@ -968,13 +968,16 @@ def ravel(a, order="C"):
 
 @_wraps(onp.squeeze)
 def squeeze(a, axis=None):
+  msg = "cannot select an axis to squeeze out " \
+            "which has size not equal to one"
   if 1 not in shape(a):
```

Comment:
this solution is a little incomplete: because of the "if 1 not in shape(a):" block, it will miss wrong axis specifications when there's no singleton dimension at all